src/net/larse/lcms/algorithms/VCT.java

```diff
@@ -0,0 +1,1567 @@
+    private double fiRange;        // Range of forest index (= uRange)
+    private double globalUdR2;     // R2 of entire series in ud B5
+    private double globalUdSlp;    // Slope of entire series in ud B5
+
+    // Attributes associated with maximum forest length segment - used
+    // in thresholding
+    private int maxConsFor;        // Maximum consecutive forest length
+    private double[] meanForUdBx;  // Mean of index values across maxConsFor
+    private double[] sdForUdBx;    // SD of index values across maxConsFor
+
+    // Chararacterization of segments based on thresholded values
+    private int[] cstSeg;          // Segment characterization
+    private int[] cstSegSmooth;    // Smoothed characterization
+    private int hudSeg;            // Number of high ud segments
+    private int ludSeg;            // Number of low ud segments
+    private int sharpTurns;        // Number of sharp turns between segments
+
+    // Disturbance variables
+    private int numDist;           // Number of disturbances detected
+    private int numMajorDist;      // Number of major disturbances detected
+    private int longestDistLength; // Length of the longest disturbance
+    private double longestDistR2;  // R2 of the longest disturbance
+    private double longestDistRough; // Roughness of the longest disturbance
+    private int[] distFlag;        // Type of each disturbance
+    private int[] distYear;        // Year of each disturbance
+    private int[] distLength;      // Length of each disturbance
+    private double[] distR2;       // R2 of each disturbance
+    private double[] distMagn;     // Magnitude (in ud composite) of each dist.
+    private double[] distMagnB4;   // Magnitude (in ud B4) of each dist.
+    private double[] distMagnVi;   // Magnitude (in ud NDVI) of each dist.
+    private double[] distMagnBr;   // Magnitude (in ud DNBR) of each dist.
+
+    // Regrowth variables
+    private double[] regrR2;       // R2 of each regrowth
+    private double[] regrSlope;    // Slope of each regrowth
+    private double[] regrRough;    // Roughness of each regrowth
+    private int[] regrFlag;        // Type of each regrowth
+
+    // Land cover/change types
+    private int lcType;            // Final land cover/change type
+    private int agIndicator;       // Agriculture indicator
+    // </editor-fold>      
+
+    public VCTSolver() {
+      this.maxUd = 4.0;
+      this.minNdvi = 0.45;
+      allocateArrays();
+    }
+
+    public VCTSolver(double maxUd, double minNdvi) {
+      this.maxUd = maxUd;
+      this.minNdvi = minNdvi;
+      allocateArrays();
+    }
+
+    /**
+     * Allocate space for all arrays so we can reuse these containers for all
+     * pixels. This is predicated on setting a maximum number of possible years
+     * that we will hold for any pixel. Each pixel's input arrays will actually
+     * determine how much of these arrays get used.
+     */
+    private void allocateArrays() {
```

Comment:
I think I'm confused then.  We had discussed allocating arrays once at the constructor and then reusing them on each call to getResult.  So at this point, we don't know the year depth, do we?  Wouldn't we have to get the find the max depth across all pixels in order to correctly allocate (other than `MAX_YEARS`)?  I'm very well not understanding.